### PR TITLE
Seeds Rework #174

### PR DESCRIPTION
**Pull request by nelsonic**
In order to run the seeds on Fly.io #172 / #173 we needed to move the bulk of the code to `/lib` that's what this PR does.
I've tried to minimise the changes but had to update a few tests to keep it all 100%. See: #174

### CODE REVIEW

File: .env_sample [hunk 1]
```diff
@@ -1,9 +1,15 @@
 export ADMIN_EMAIL=youremail@gmail.com
-export AUTH_API_KEY=2PzB7PPnpuLsbWmWtXpGyI+kfSQSQ1zUW2Atz/+8PdZuSEJzHgzGnJWV35nTKRwx/dwylauth.herokuapp.com
-export EMAIL_APP_URL=https://dwylmail.herokuapp.com
+export AUTH_URL=dwylauth.herokuapp.com
```
Comment (nelsonic) [inline .env_sample:2]: Had to add the `AUTH_URL` environment variable to avoid the circular dependency on `auth_plug` noted by @tadasajon in #170

**Comment by nelsonic** [comment]
@SimonLab please take a look at this PR and LMK what you think. Feedback very much welcome!

File: priv/repo/seeds.exs [hunk 2]
```diff
@@ -1,147 +1,2 @@
...
-# Script for populating the database. You can run it as:
-#
-# mix run priv/repo/seeds.exs
-#
-# The seeds.exs script will also run
-# when you execute the following command
-# to setup the database:
-#
-# mix ecto.setup
-defmodule Auth.Seeds do
```
Comment (nelsonic) [inline priv/repo/seeds.exs:11]: The bulk of this code moved to [`init.ex`](https://github.com/dwyl/auth/pull/175/files#diff-d106602d01c212adbdc5cc818402766b0907b01a9a0da2bfea97e97e4e8d912b) above. ⬆️

File: lib/auth/init/init.ex [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  """
+
+  require Logger
+  import Ecto.Changeset
+  alias Auth.{Person, Role, Repo, Status}
+
+  def main do
+    Logger.info("Initialising the Auth Database ...")
+    # check required environment variables:
+    Envar.is_set_all?(~w/ADMIN_EMAIL AUTH_URL ENCRYPTION_KEYS SECRET_KEY_BASE/)
```
Comment (SimonLab) [inline lib/auth/init/init.ex:27]: We could create the list of environment variables as a module variable to make it easier to change later on if required. However I think it does the job at the moment

File: lib/auth/init/init.ex [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+      # ignore the next lines because we can't test them:
+      # coveralls-ignore-start
+      _ ->
+        # Log the AUTH_API_KEY so it can be exported:
+        Logger.info("export AUTH_API_KEY=#{api_key}")
+        # coveralls-ignore-stop
+    end
+
+    # Update status of Admin to "verified"
+    Auth.Person.verify_person_by_id(1)
```
Comment (SimonLab) [inline lib/auth/init/init.ex:50]: Can we use:
```suggestion
    Auth.Person.verify_person_by_id(admin.id)
```
To make it clearer that the id 1 is the admin we just created?

**Review by SimonLab — COMMENTED** [review]
Looks good to me, I've just added a couple of comments, let me know what you think @nelsonic but happy to merge otherwise

**Comment by nelsonic** [comment]
@SimonLab thanks very much for reviewing. I've applied your suggested change to `init.ex` ✅ 
Please Merge if/when you're happy. Thanks. 👍

**Review by SimonLab — APPROVED** [review]

**Pull request merged**